File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.java
import java.io.*;
import java.time.*;
import java.time.format.DateTimeFormatter;
import java.util.*;

public class Main {
	static int board[][],dist[][];
	static int n,m;
	static int INT_MAX = Integer.MAX_VALUE;
	static StringTokenizer st;
	
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		
		board = new int[n+1][n+1];
		dist = new int[n+1][n+1];
		for(int y=1; y<=n; y++) {
			st = new StringTokenizer(br.readLine());
			for(int x=1; x<=n; x++) {
				board[y][x] = Integer.parseInt(st.nextToken());
				dist[y][x] = board[y][x];
			}
		}
		
		for(int k =1 ; k<=n; k++) {
			for(int i =1; i<=n; i++) {
				for(int j = 1;j<=n;j++) {
					if(dist[i][j] <= dist[i][k] + dist[k][j])
						continue;
					dist[i][j] = Math.min(dist[i][j], dist[i][k] + dist[k][j]);
				}
			}
		}
		
		StringBuilder sb = new StringBuilder();
		for(int i = 0 ; i < m; i++) {
			st = new StringTokenizer(br.readLine());
			int a = Integer.parseInt(st.nextToken());
			int b = Integer.parseInt(st.nextToken());
			int cost = Integer.parseInt(st.nextToken());
			
			sb.append(dist[a][b] <= cost ? "Enjoy other party" : "Stay here").append('\n');
		}
		System.out.println(sb);
		
		
		
		
	}
}

File: 백준/Gold/11265. 끝나지 않는 파티/끝나지 않는 파티.py

if __name__ == "__main__":
    N,M = map(int,input().split())
    board = [[] for _ in range(N+1)]

    for y in range(1,N+1):
        board[y] = [0] + list(map(int,input().split()))

    for k in range(1, N+1):
        for y in range(1, N+1):
            for x in range(1, N+1):
                if board[y][x] > board[y][k] + board[k][x]:
                    board[y][x] = board[y][k] + board[k][x]

    ans = []
    for _ in range(M):
        a,b,cost = map(int,input().split())

        ans.append("Enjoy other party" if board[a][b] <= cost else "Stay here")

    print('\n'.join(ans))